File: frontend/src/types/campaign.ts
export type Channel = "TWITTER" | "YOUTUBE" | "PINTEREST";
export type CampaignStatus = "DRAFT" | "ACTIVE" | "PAUSED" | "COMPLETED";

export interface Campaign {
  id: number;
  productId: number;
  name: string;
  channel: Channel;
  dailyLimit: number; // integer
  startDate?: string | null; // ISO date (yyyy-mm-dd)
  endDate?: string | null;   // ISO date or null
  status: CampaignStatus;
  config?: Record<string, any> | null; // channel-specific
  createdAt?: string;
}

export interface CampaignCreateRequest {
  productId: number;
  name: string;
  channel: Channel;
  dailyLimit: number;
  startDate?: string | null;
  endDate?: string | null;
  config?: Record<string, any> | null;
}

export interface ChannelType {
  name: string;
  displayName: string;
  description: string;
  configFields: string[];
}

File: backend/campaign-service/src/main/java/com/atb/campaignservice/entity/Campaign.java
package com.atb.campaignservice.entity;

import com.atb.campaignservice.enums.CampaignStatus;
import com.atb.campaignservice.enums.Channel;
import io.hypersistence.utils.hibernate.type.json.JsonBinaryType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Type;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Map;

@Entity
@Table(name = "campaigns")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Campaign {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long productId;

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Channel channel;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = true)
    private LocalDate endDate;

    @Column(nullable = false)
    private Integer dailyLimit;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private CampaignStatus status = CampaignStatus.DRAFT;

    @Type(JsonBinaryType.class)
    @Column(columnDefinition = "jsonb")
    private Map<String, Object> config;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (status == null) {
            status = CampaignStatus.DRAFT;
        }
    }
}

